packages/typescript/internal/ts_project_options_validator.ts

```diff
@@ -93,6 +93,28 @@ function main([tsconfigPath, output, target, attrsStr]: string[]): 0|1 {
     console.error('- Otherwise, remove the noEmit option from tsconfig and try again.');
     return 1;
   } 
+  
+  // When there are dependencies on other ts_project targets, the tsconfig must be configured
+  // to help TypeScript resolve them.
+  if (attrs.has_local_deps) {
+    let rootDirsValid = true
+    if (!options.rootDirs) {
+      console.error(`ERROR: ts_project rule ${target} is configured without rootDirs.`);
+      rootDirsValid = false;
+    }
+    else if (!options.rootDirs.some(d => d.startsWith(process.env['BINDIR']))) {
+      console.error(`ERROR: ts_project rule ${target} is missing a needed rootDir under ${process.env['BINDIR']}.`);
+      console.error('Found only: ', options.rootDirs);
+      rootDirsValid = false;
+    }
+    if (!rootDirsValid) {
+      console.error('This makes it likely that TypeScript will be unable to resolve dependencies using relative import paths');
+      console.error(`For example, if you 'import {} from ./foo', this expects to resolve foo.d.ts from Bazel's output tree`);
+      console.error('and TypeScript only knows how to locate this when the rootDirs attribute includes that path.');
+      console.error('See the ts_project documentation.');
```

Comment:
nit: link to the docs?

Reply:
I'm so scared about what's a valid syntax for the link that will work in all the places...